client/src/components/AddFriendPage.js

```diff
@@ -0,0 +1,112 @@
+import React, { useContext, useState, useEffect} from 'react';
+import { GlobalProvider, GlobalContext } from '../context/GlobalState';
+import { FriendList } from './Chat/FriendList';
+import { Link } from 'react-router-dom';
+import { userActions } from '../context/_actions';
+import { useDispatch, useSelector } from 'react-redux';
+
+export const AddFriendPage = () => {
+	
+	const [inputs, setInputs] = useState({
+        friendUsername: ''
+    });
+    const [submitted, setSubmitted] = useState(false);
+    const { friendUsername} = inputs;
+    const users = useSelector(state => state.users);
+    const user = useSelector(state => state.authentication.user);
+    const { friends, getFriends, addFriend } = useContext(GlobalContext);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(userActions.getAll());
+    }, [dispatch]);
+
+
+    function handleChange(e) {
+        const { name, value } = e.target;
+        setInputs(inputs => ({ ...inputs, [name]: value }));
+    }
+
+    function findFriend(friendusername){
+    	for(let i=0; i<user.friends.length; i++){
+    		if(user.friends[i].username === friendusername){
+    			return false;
```

Comment:
as the function is findFriend, I think it is more reasonable to let it return true when you find the given name in the friend list.